app/src/main/java/com/geeksville/mesh/service/MeshService.kt

```diff
@@ -942,22 +943,22 @@ class MeshService : Service(), Logging {
         val p = packet.decoded
 
         // If the rxTime was not set by the device (because device software was old), guess at a time
-        val rxTime = if (packet.rxTime == 0) packet.rxTime else currentSecond()
+        val rxTime = if (packet.rxTime != 0) packet.rxTime else currentSecond()
```

Comment:
@geeksville : I think this was not good if rxTime was 0 if not present in the packet

Reply:
@lgoix hmm - I don't quite understand.  if rxTime is not specified by the device, why would we not want to pick the local phone time?

Reply:
Yes that is the fix

It was: if rxtime is set then use the alocal phone time